superset-frontend/src/chart/ChartRenderer.jsx

```diff
@@ -78,28 +119,35 @@ class ChartRenderer extends React.Component {
   }
 
   shouldComponentUpdate(nextProps) {
-    const resultsReady =
-      nextProps.queryResponse &&
-      ['success', 'rendered'].indexOf(nextProps.chartStatus) > -1 &&
-      !nextProps.queryResponse.error &&
-      !nextProps.refreshOverlayVisible;
-
-    if (resultsReady) {
-      this.hasQueryResponseChange =
-        nextProps.queryResponse !== this.props.queryResponse;
-
-      if (
-        this.hasQueryResponseChange ||
-        nextProps.annotationData !== this.props.annotationData ||
-        nextProps.height !== this.props.height ||
-        nextProps.width !== this.props.width ||
-        nextProps.triggerRender ||
-        nextProps.formData.color_scheme !== this.props.formData.color_scheme
-      ) {
-        return true;
-      }
+    this.hasQueryResponseChange =
+      this.props.queryResponse !== nextProps.queryResponse;
+
+    // if no results loaded, don't render
+    if (
+      !nextProps.queryResponse ||
+      nextProps.queryResponse.error ||
+      nextProps.refreshOverlayVisible
+    )
+      return false;
+
+    // current chart status
+    const chartStatus = this.props.chartStatus;
+    // `rendered` status is set right after `success` or `loading`,
+    // don't trigger rerender here (see `actions.chartRenderingSucceeded`).
+    // note that even though render is not triggered, the props will still be
+    // updated.
```

Comment:
Not sure what we are using the `rendered` status for, but I'm keeping it here just in case.

Reply:
The logic seems to be changed here, since the original code did not check `this.props.chartStatus`.